ark/utils/load_utils.py

```diff
@@ -319,3 +321,197 @@ def load_imgs_from_dir(data_dir, files=None, match_substring=None, trim_suffix=N
+
+    Args:
+        data_dir (str):
+            directory containing folders of images
+        img_sub_folder (str):
+            optional name of image sub-folder within each fov
+        channels (list):
+            optional list of imgs to load, otherwise loads all imgs
+        max_image_size (int or None):
+            The length (in pixels) of the largest image that will be loaded. All other images will
+            be padded to bring them up to the same size.
+
+    Returns:
+        xarray.DataArray:
+            xarray with shape [fovs, x_dim, y_dim, tifs]
+    """
+
+    iou.validate_paths(data_dir, data_prefix=False)
+
+    if img_sub_folder is None:
+        # no img_sub_folder, change to empty string to read directly from base folder
+        img_sub_folder = ''
+
+    fov_list = ns.natsorted(iou.list_folders(data_dir))
+    if 'stitched_images' in fov_list:
+        fov_list.remove('stitched_images')
+
+    # check for RnCm folder naming
+    for dir in fov_list:
+        r = re.compile('.*R.*C.*')
+        if r.match(dir) is None:
+            raise ValueError(f"Invalid FOVs found in directory, {data_dir}. FOV folder names "
+                             f"should have the form RnCm.")
+    if len(fov_list) == 0:
+        raise ValueError(f"No FOVs found in directory, {data_dir}.")
+
+    expected_fovs = get_tiled_fov_names(fov_list)
+
+    # get image size if not provided
+    if not max_image_size:
+        max_image_size = get_max_img_size(data_dir, img_sub_folder, fov_list)
+
+    # get imgs from first fov if no img names supplied
+    if channels is None:
+        channels = iou.list_files(
+            dir_name=os.path.join(data_dir, fov_list[0], img_sub_folder),
+            substrs=['.tif', '.jpg', '.png']
+        )
+        channels.sort()
+    # otherwise, fill channel names with correct file extension
+    elif not all([img.endswith(("tif", "tiff", "jpg", "png")) for img in channels]):
+        # need this to reorder channels back because list_files may mess up the ordering
+        channels_no_delim = [img.split('.')[0] for img in channels]
+
+        all_channels = iou.list_files(
+            dir_name=os.path.join(data_dir, fov_list[0], img_sub_folder),
+            substrs=channels_no_delim, exact_match=True
+        )
+
+        # get the corresponding indices found in channels_no_delim
+        channels_indices = [channels_no_delim.index(chan.split('.')[0]) for chan in all_channels]
+
+        # verify if channels from user input are present in `all_channels`
+        all_channels_no_delim = [channel.split('.')[0] for channel in all_channels]
+
+        misc_utils.verify_same_elements(all_channels_in_folder=all_channels_no_delim,
+                                        all_channels_detected=channels_no_delim)
+        # reorder back to original
+        channels = [chan for _, chan in sorted(zip(channels_indices, all_channels))]
+
+    if len(channels) == 0:
```

Comment:
Looks like a lot of the logic in this function is repeated in the main function, you should decide on the logical place for these checks, and if any of them need to be in both.